File: leet_code/136/2.go
package main

import "fmt"

func singleNumber(nums []int) int {
	// return 0
	m := make(map[int]int)
	for _, v := range nums {
		// fmt.Println(v)
		_, ok := m[v] //检测 hashmap 有这个k吗
		// fmt.Println(ok)
		if ok { //出现过
			delete(m, v)
		} else {
			m[v] = 1
		}
	}
	for k := range m {
		// fmt.Println(k)
		return k
	}

	return 0
}
func main() {
	fmt.Printf("%d", singleNumber([]int{1, 4, 1, 2, 2}))
}

File: leet_code/136/2.js
var singleNumber = function(nums) {
  var map = new Map()
  var N 
  nums.forEach(i => {
    map.set(i, map.has(i)? map.get(i)+1:1)
  })
  // console.log(map)

  map.forEach(function (key, value) {
    if (key === 1)
    {
      N = value
    }
    // console.log(value, key)
  })
  return N
};
console.log(singleNumber([4,1,2,1,2]))